### DISCUSSION TITLE
Updated from 7.7.0 to 8.0.0 and now cannot connect (or it disconnects after a minute and never reconnects)

### DISCUSSION THREAD

**Discussion by Candurin** [tagyoureit/nodejs-poolController#776] [App and O/S Config Questions]
I had no luck in getting any sort of logs to work.  Here is the output when I try and start pool controller and dash panel:

Pool Controller:

pi:~/nodejs-poolController/data $ 
> nodejs-poolcontroller@8.0.1 start
> npm run build && node dist/app.js

> nodejs-poolcontroller@8.0.1 build
> tsc

[6/5/2023, 3:39:44 PM] info: Starting Replay Capture.
Init state for Pool Controller
[6/5/2023, 3:39:44 PM] info: The current git branch output is master
[6/5/2023, 3:39:44 PM] info: The current git commit output is 64743d6c4b194c1a5cb7e5ea3d991f3e77844089
[6/5/2023, 3:39:44 PM] info: Starting up SSDP server
Error: listen EADDRINUSE: address already in use 0.0.0.0:4200
    at Server.setupListenHandle [as _listen2] (node:net:1740:16)
    at listenInCluster (node:net:1788:12)
    at doListen (node:net:1937:7)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)

DashPanel:

pi:~/nodejs-poolController-dashPanel $ npm start&
[1] 2909
pi@pi:~/nodejs-poolController-dashPanel $ 
> nodejs-poolcontroller-dashpanel@8.0.0 start
> npm run build && node dist/app.js

> nodejs-poolcontroller-dashpanel@8.0.0 build
> tsc

Updated configuration file
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use 0.0.0.0:5150
    at Server.setupListenHandle [as _listen2] (node:net:1740:16)
    at listenInCluster (node:net:1788:12)
    at doListen (node:net:1937:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1767:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -98,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 5150
}

Node.js v18.16.0

Note: Everything was working fine in 7.7.0 but all broke after my update.  Any help would be appreciated.

**Comment by tagyoureit** [comment]
The EADDRIINUSE means you already have an instance of dashPanel (at port 5150) and njsPC (at port 4200) already running.  Either you didn't stop/restart the services before updating them or you created another parallel installation and started both 7.7 and 8.0 at the same time.

**Reply by Candurin** [reply]
So what should I do to fix this?  Just scrap everything and start anew?

Or how can I just delete all of the 7.7.0?  I'd hate to start over.

**Reply by tagyoureit** [reply]
How did you "upgrade"?  If you did a `git pull` and updated from 7.7 to 8.0 a simple reboot might fix everything.  If you installed in multiple directories and have multiple startup scripts you will need to remove one of the copies.  The best thing to do is an in place upgrade which will keep your existing data and config files (otherwise you would need to move them to the new directory or start over).

**Reply by Candurin** [reply]
rstrouse had me do a few things over at TFP forum.   Where am I looking for multiple copies.  Sadly, my linux knowledge is minimal.  I can certainly back up my config files (from the GUI in RPi) but that's about it.  Sorry!

**Reply by tagyoureit** [reply]
You can install [htop](https://itsfoss.com/use-htop/) and it will show you the locations of the two processes.  

Did you try rebooting and do you have the processes set to automatically start?

**Reply by Candurin** [reply]
I have tried rebooting and yes, I am using PM2 to auto start the processes.

**Reply by tagyoureit** [reply]
If you are using PM2 than you can't use `nodejs-poolController-dashPanel $ npm start&` at the same time.  This will start a second instance of dashPanel (or respectively, njsPC if you run the command in that directory as well).

**Reply by Candurin** [reply]
Do you recommend using PM2 or creating the local startup file with the nom start& commands?

At this point I have deleted and reinstalled everything.  Now I can't get the connection to NOT disconnect.

**Reply by Candurin** [reply]
Seems like this is all related to the fact nothing is booting at startup.  If I start each manually, it's fine:

pi        1797  2.0  5.3 171080 50824 pts/0    Sl   21:56   0:04 npm start
pi        1812  0.0  0.0   1940   360 pts/0    S    21:56   0:00 sh -c npm run build && node dist/app.js
pi        2093  4.0  5.4 170912 51352 pts/1    Sl   21:58   0:04 npm start
pi        2104  0.0  0.0   1940   368 pts/1    S    21:58   0:00 sh -c npm run build && node dist/app.js
pi        2393  0.0  0.0   7344   576 pts/2    S+   22:00   0:00 grep --color=auto npm

But nothing is running when I reboot, despite my rc.local file being correct:

fi

cd ~/nodejs-poolController

npm start&​

cd ~/nodejs-poolController-dashPanel

npm start&​

exit 0

Please note that I deleted "pi" from the directory names (cd ~pi/nodejs-poolController-dashPanel) as my directory structure is as above.  Additionally, I have tried with "pi" in the directory and it makes no difference

**Reply by tagyoureit** [reply]
See https://github.com/tagyoureit/nodejs-poolController/wiki/Automatically-start-at-boot---PM2-&-Systemd for the PM2 instructions and another method.

**Reply by Candurin** [reply]
Just wanted to thank you for the help.  Switched to PM2 and everything working perfectly!
